api/users.go

```diff
@@ -125,16 +126,27 @@ func (r usersRoutes) updateUser(c *gin.Context) {
 
 func (r usersRoutes) prepareUserSearch(c *gin.Context) (users []model.User, err error) {
 	q := c.Query("q")
-	reg, _ := regexp.Compile("[^a-zA-Z0-9 ]+")
+	rQ := c.Query("r")
+	reg := regexp.MustCompile("[^a-zA-Z0-9 ]+")
 	q = reg.ReplaceAllString(q, "")
-	if len(q) < 3 {
+	// make the search work with empty query but selected role
+	if len(q) < 3 && (rQ == "-1" || rQ == "") {
 		_ = c.Error(tools.RequestError{
 			Status:        http.StatusBadRequest,
 			CustomMessage: "query too short (minimum length is 3)",
 		})
 		return nil, errors.New("query too short (minimum length is 3)")
 	}
-	users, err = r.UsersDao.SearchUser(q)
+	role, err := strconv.ParseUint(rQ, 10, 64)
+	if err != nil && rQ != "" && rQ != "-1" {
+		tools.RenderErrorPage(c, http.StatusBadRequest, "invalid role")
```

Comment:
API endpoints should not render HTML pages on error but rather use 
https://github.com/TUM-Dev/gocast/blob/95804eb169b5a717f715c55fe6858478c098ccec/web/course.go#L31